src/arch/ArchHooks/ArchHooks_MacOSX.mm

```diff
@@ -258,19 +258,16 @@ static bool DoEmergencyShutdown( int signal, siginfo_t *si, const ucontext_t *us
 	return result == 0;
 }
 
-std::int64_t ArchHooks::GetMicrosecondsSinceStart( bool bAccurate )
+std::uint64_t ArchHooks::GetMicrosecondsSinceStart( bool bAccurate )
 {
-	// http://developer.apple.com/qa/qa2004/qa1398.html
-	static double factor = 0.0;
-
-	if( unlikely(factor == 0.0) )
-	{
-		mach_timebase_info_data_t timeBase;
-
-		mach_timebase_info( &timeBase );
-		factor = timeBase.numer / ( 1000.0 * timeBase.denom );
-	}
-	return std::int64_t( mach_absolute_time() * factor );
+    static std::uint64_t factor = 0;
```

Comment:
Aren't we going to lose precision if we go from a double to an integer?

`(timeBase.numer / timeBase.denom) / 1000` will be a whole number instead of some decimal value.

Reply:
Ahh yes, thank you for catching this.